src/System.Net.Sockets/src/System/Net/Sockets/Socket.Windows.cs

```diff
@@ -122,16 +122,16 @@ internal unsafe bool TransmitPackets(SafeCloseSocket socketHandle, IntPtr packet
             return transmitPackets(socketHandle, packetArray, elementCount, sendSize, overlapped, flags);
         }
 
-        internal static IntPtr[] SocketListToFileDescriptorSet(IList socketList)
+        internal static void SocketListToFileDescriptorSet(IList socketList, Span<IntPtr> fileDescriptorSet)
         {
-            if (socketList == null || socketList.Count == 0)
+            int count;
+            if (socketList == null || (count = socketList.Count) == 0)
             {
-                return null;
+                return;
             }
 
```

Comment:
Nit: consider adding an assert for `fileDescriptorSet.Length >= socketList.Count`, servicing two purposes.  In addition to the assert itself, it'll help to highlight for someone reading the code that fileDescriptorSet may actually be bigger than socketList.

Reply:
adding; it is actually **strictly** greater (although >= socketList.Count + 1 may more intuitively remind the reader about the prefix element)